synapse/config/repository.py

```diff
@@ -137,6 +137,8 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
         self.max_image_pixels = self.parse_size(config.get("max_image_pixels", "32M"))
         self.max_spider_size = self.parse_size(config.get("max_spider_size", "10M"))
 
+        self.prevent_downloads_from = config.get("prevent_downloads_from", [])
```

Comment:
Should we do any validation of this? Should we at least prevent people from putting their local domain in here?

Reply:
I personally believe the documentation warning is enough. Ignoring the docs and doing silly things warrants the adverse behaviour in my opinion :p